tests/unit/package_managers/test_npm.py

```diff
@@ -615,17 +615,32 @@ def test_fetch_npm_source(
     assert output == expected_output
 
 
-@mock.patch("pathlib.Path.exists")
-def test_resolve_npm_no_lock(
+@pytest.mark.parametrize(
+    "lockfile_exists, node_mods_exists, expected_error",
+    [
+        pytest.param(
+            False,
+            False,
+            "The npm-shrinkwrap.json or package-lock.json file must be present for the npm package manager",
+            id="no lockfile present",
+        ),
+        pytest.param(
+            True,
+            True,
+            "The 'node_modules' directory cannot be present in the source repository",
+            id="lockfile present; node_modules present",
+        ),
+    ],
+)
+@mock.patch("pathlib.Path.exists", autospec=True)
+def test_resolve_npm_basic(
```

Comment:
Nitpick: I'd probably rename this test to `test_resolve_npm_errors` or `test_resolve_npm_validation`.